crawl-ref/source/player.cc

```diff
@@ -7940,7 +7940,7 @@ vector<PlaceInfo> player::get_all_place_info(bool visited_only,
 // different effect from the player invokable ability.
 bool player::do_shaft()
 
-    if (!is_valid_shaft_level()) || (species == SP_FORMICID)
+    if (!is_valid_shaft_level()) || if (species == SP_FORMICID)
```

Comment:
This should not be here, but instead in traps.cc: 938. Check out the code in that line and you can probably figure it out. You want you.species == SP_FORMICID in that conditional.